internal/abci/abci.go

```diff
@@ -594,6 +672,16 @@ func (a *AbciApp) FinalizeBlock(ctx context.Context, req *abciTypes.RequestFinal
 		res.ValidatorUpdates[i] = abciTypes.Ed25519ValidatorUpdate(up.PubKey, up.Power)
 	}
 
+	// wait for all logs to be received
+	wg.Wait()
+	for _, result := range resultArr {
+		logs, ok := logMap[hex.EncodeToString(result.TxHash)]
```

Comment:
The map key is a postgresql transaction id from `txid_current`, but this is trying to access by the blockchain transaction hash.

Reply:
I might be mixing up something, or the PR is mixing concepts from the first iteration.  I see that parse defines `PGFormat` with `return fmt.Sprintf("notice('txid:' || current_setting('ctx.txid') || ' ' || %s)", inputs[0]), nil`.
So does this mean that we actually have to get the _blockchain_ txid from the log text, which would be _after_ the pg txid in the text?
So, for each tx result, check for a some log text that includes that chain txid in the logged notice?

Reply:
Arg, nvm, I misread `ParseNotice`.  It's getting the blockchain txid as I would expect it should.